vac/raw/consensus-hashgraphlike.md

```diff
@@ -0,0 +1,212 @@
+  bool liveness_criteria_yes = 18;  // Shows how managing the silent peers vote
+}
+
+message Vote {
+  int32 vote_id = 20;             // Unique identifier of the vote
+  bytes vote_owner = 11;          // Voter's public key
+  int64 timestamp = 22;           // Time when the vote was cast
+  bool vote = 23;                 // Vote bool value (true/false)
+  bytes parent_hash = 24;         // Hash of previous owner's Vote
+  bytes received_hash = 25;       // Hash of previous received Vote
+  bytes vote_hash 26 =            // hash of Vote hash(vote_id, owner, timestamp, 
+  vote, parent_hash, received_hash)
+  bytes signature = 27;           // Signature of hash Sign (hash)
+}
+
+```
+
+In particular, to initialize a consensus for a proposal,
+a user MUST fulfill all the fields in the proposal including attaching its vote.
+In particular, parent_hash and received_hash are empty string since there is no previous or received hash.
+Then the initialization section ends when the user who creates the proposal sends it
+to the random peer from the network or sends it to the proposal to the specific channel.
+
+## 2. Exchanging votes across the peers
+
+Once the peer receives the proposal message P_1 from a 1-1 or a gossipsub channel with its votes does the following checks:
+
+- Check the signatures of the voting(should it be the recursive from the first one to the last one)
+- Check the hash checks in particular:
+  - Parenthash check prevents double voting
+  - Receivedhash check provides tampering attacks
+
+In particular, proposal message P_1 where its vote V_1 = P_1.votes[0] and performs as follows:
+
+- Verifiy the vote V_1.signature with with the V_1.vote_owner in the proposal.
+- If the peer verifies the signature,
+it continues to create P_2 with the new vote V_2 that consists of as following:
+  - adding its public key
+  - timestamp
+  - boolean vote
+  - V_2.parent_hash = 0 if there is no previous peer's vote, otherwise Hash of previous owner's vote
+  - V_2.received_hash = P_1.votes[0]
+  - Calculate vote_hash by  hash of Vote hash(vote_id, owner, timestamp, vote, parent_hash, received_hash)
+    then adds the V_2.vote_hash
+  - Sign vote_hash with its private key then adds V_2.vote_hash.
+- Create P_2 with by adding V_2 as follows:
+  - P_2.name, P_2.proposal_id and P_2.proposal_owner are the same with P_1.
+  - Add the V_2 to the P_2.Votes list.
+  - Increase the round by one, namely P_2.round = P_1.round + 1.
+  - Verify the time proposal timestamp is valid for expiration time, namely P_2.timestamp - current < P_1.expiration_time.
+  If this does not hold, other peers ignore the message.
+
+After the peer creates the proposal P_2 with its vote V_2,
+sends it to the random peer from the network or
+sends it to the proposal to the specific channel.
+
+## 3. Determining the Result
+
+Since a peer cannot know if the votes are enough for the consensus without counting,
+it requires checking each time by the peers.

```

Comment:
sounds like: we can't understand without counting, so we count it 

is it right understanding?